src/executables/sysrepo-plugind.c

```diff
@@ -567,6 +602,30 @@ main(int argc, char **argv)
     sd_notify(0, "READY=1");
 #endif
 
+    /* update pid file (if daemon) */
+    if (!debug && pidfile) {
+        int pid_len;
+        if (ftruncate(pidfd, 0)) {
+            error_print(0, "Failed to truncate PID file (%s).", strerror(errno));
+            goto cleanup;
+        }
+
+        if (asprintf(&pid, "%d\n", getpid()) < 0) {
+            error_print(0, "Failed to allocate memory for pid (%s).", strerror(errno));
+            goto cleanup;
+        }
+
+        pid_len = (int) strlen(pid);
+        if (write(pidfd, pid, pid_len) < pid_len) {
+            error_print(0, "Failed to write into PID file (%s).", strerror(errno));
+            goto cleanup;
+        }
+        close(pidfd);
+        pidfd = -1;
+        free(pid);
+        pid = NULL;
```

Comment:
why not rely on the `cleanup`?

Reply:
Well, I figured that there's no need to hang on to memory if it's not needed. Old habit... of course, this is a very small chunk of memory so it doesn't really matter in most systems.